src/lib.rs

```diff
@@ -387,6 +437,45 @@ mod tests {
         })
     }
 
+    /// [`store_path_perms`] stores [`ExampleConfig`], with only read permission for owner (UNIX).
+    #[test]
+    #[cfg(unix)]
+    fn test_store_path_perms() {
+        with_config_path(|path| {
+            let config: ExampleConfig = ExampleConfig {
+                name: "Secret".to_string(),
+                count: 16549,
+            };
+            store_path_perms(path, &config, Permissions::from_mode(0o600))
+                .expect("store_path_perms failed");
+            let loaded = load_path(path).expect("load_path failed");
+            assert_eq!(config, loaded);
+        })
+    }
+
+    /// [`store_path_perms`] stores [`ExampleConfig`], as read-only.
+    #[test]
+    fn test_store_path_perms_readonly() {
+        with_config_path(|path| {
+            let config: ExampleConfig = ExampleConfig {
+                name: "Soon read-only".to_string(),
+                count: 27115,
+            };
+            store_path(path, &config).expect("store_path failed");
+
```

Comment:
Unfortunately there is a nit about this nit : test will fail on (UNIX) hosts with an umask of `0o222` (or "above"), as new files will be read-only by default.

Reply:
Even between 2 stores ? I’ll have to believe you there.

I’m not sure what you test with 2 stores one after the other like that. The first store (a line above)
https://github.com/rust-cli/confy/blob/9b6da7929e98c5676a310a4900f8c0d0f8b978ae/src/lib.rs#L464
should open, write the file and close it.

So at that point the file is closed and you want to write to it again, but setting permissions.

Maybe you did not want to store to file twice? (That is also a good "fix" to only test 1 store : the perm one.)

Anyway to me the PR is good, I learned some too :)

Reply:
> Even between 2 stores ? I’ll have to believe you there.

On an UNIX host :

1. Storing without specifying permissions with an umask of `0o222` causes files to miss writing bits --> `readonly` attribute will evaluate to `false`, hence adding such a test would fail.
2. Storing and specifying `readonly` will always succeed (no umask could prevent this) --> proposed test is environment-safe.

However, I don't have your project testing policy in mind, and maybe you don't expect unit tests to pass in such environments.

> Maybe you did not want to store to file twice? (That is also a good "fix" to only test 1 store : the perm one.)

I did want to, but Rust standard library does not allow `std::fs::Permissions` direct instantiation, so I've opted for a snippet where we could safely switch the `readonly` attribute from metadata read from a file descriptor we know we could open in this testing environment.
I could have modified `test_store_path` case but didn't want to update existing tests.

Thanks for your time, bye :wave: